surrealdb/ws.py

```diff
@@ -219,21 +220,48 @@ async def __aexit__(
         """
         await self.close()
 
-    async def connect(self) -> None:
-        """Connect to a local or remote database endpoint.
+    async def connect(self, url: Optional[str] = None, max_size: Optional[int] = None) -> None:
+        """Connects to a local or remote database endpoint.
+
+        Args:
+            url: The URL of the Surreal server.
 
         Examples:
             Connect to a local endpoint
                 db = Surreal()
                 await db.connect('ws://127.0.0.1:8000/rpc')
                 await db.signin({"user": "root", "pass": "root"})
+
+            Connect to a remote endpoint
+                db = Surreal()
+                await db.connect('http://cloud.surrealdb.com/rpc')
+                await db.signin({"user": "root", "pass": "root"})
         """
-        self.ws = await websockets.connect(self.url)  # type: ignore
+        if url is not None:
+            self.url = url
+        else:
+            self.url
+        if max_size is not None:
+            self.max_size = max_size
+        else:
+            # default max size is 1MB
+            self.max_size = 2**20
+        # helping people when they type the url in wrong
+        if "http" in self.url:
+            self.url = self.url.replace("http://", "ws://")
+        elif "https" in self.url:
+            self.url = self.url.replace("https://", "wss://")
+        if "/rpc" not in self.url:
+            self.url = "".join([self.url, "/rpc"])
```

Comment:
This may lead some unexpected behaviour (`https://httpserver.cat` would match first case)

If something like this needs to be implemented, using something like urllib would be a better choice (https://docs.python.org/3/library/urllib.parse.html)